internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,348 @@
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    primary key(config_id, event_type)
+);
+
+comment on table event_type_enabled is
+'event_type_enabled is a table where each row represents that eventing has '
+'been enabled for the specified event type in an event configuration';
+
+create table event_sink(
+    public_id wt_public_id primary key,
+    config_id wt_public_id not null
+        constraint event_config_fkey
+            references event_config(public_id)
+            on delete cascade
+            on update cascade
+);
+
+comment on table event_sink is 
+'event_sink is a table where each row represents a configured event sink';
+
+-- insert_event_sink_subtype() is a before insert trigger
+-- function for subtypes of event_sink
+create function insert_event_sink_subtype()
+returns trigger
+as $$
+begin
+insert into event_sink
+    (public_id, sink_id)
+values
+    (new.public_id, new.sink_id);
+return new;
+end;
+$$ language plpgsql;
+
+-- delete_event_sink_subtype() is an after delete trigger
+-- function for subtypes of event_sink
+create function delete_event_sink_subtype()
+returns trigger
+as $$
+begin
+delete from event_sink
+where public_id = old.public_id;
+return null; -- result is ignored since this is an after trigger
+end;
+$$ language plpgsql;
+
+create table event_file_sink(
+    public_id wt_public_id primary key,
+    sink_id wt_public_id not null
+        constraint event_sink_fkey
+            references event_sink(public_id)
+            on delete cascade
+            on update cascade,
+    event_type text not null
+        constraint event_type_enm_fkey
+            references event_type_enm(name)
+            on delete restrict
+            on update cascade,
+    format_type text not null
+        constraint event_format_type_enm_fkey
+            references event_format_type_enm(name)
+            on delete restrict
+            on update cascade,
+    allow_filter wt_bexprfilter,
+    deny_filter wt_bexprfilter,
+    path text not null 
+        constraint path_not_empty
+            check (
+                length(trim(path)) > 0
+            ),
+    filename text not null
+        constraint filename_not_empty
+            check (
+                length(trim(filename)) > 0
+            ),
+    constraint path_filename_uq
+        unique(path, filename), -- ensure each sink is writing to a unique file
+    rotate_bytes int
+        constraint rotate_bytes_null_or_greater_than_zero
+            check(
+                rotate_bytes > 0
+            ),
+    rotate_duration interval
+        constraint rotate_duration_null_or_greater_than_zero
+            check(
+                rotate_duration > '0'::interval
+            ),
+    rotate_max_files int
+        constraint rotate_max_files_null_or_greater_than_zero
+            check(
+                rotate_max_files > 0
+            )
```

Comment:
Needs a foreign key constraint to the `event_sink` table. Example: [`credential_vault_store` table's foreign key constraint to the the `credential_store` table](https://github.com/hashicorp/boundary/blob/248e1f36b5ed3ed24a0ec92e47a2de70fdb5b4ef/internal/db/schema/migrations/postgres/10/04_vault_credential.up.sql#L30-L34).